### PR TITLE
JAVA: Key management API

### PR DESCRIPTION
**Pull request by rozza**
Added RewrapManyDataKey support and MongoRewrapManyDataKeyOptions

JAVA-4433

### CODE REVIEW

File: bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/MongoCryptImpl.java [hunk 1]
```diff
@@ -232,19 +233,27 @@ public MongoCryptContext createDataKeyContext(final String kmsProvider, final Mo
         }
 
         if (kmsProvider.equals("aws")) {
```
Comment (kevinAlbs) [inline bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/MongoCryptImpl.java:235]: Optional: You can remove the "aws" and "local" cases. `mongocrypt_ctx_setopt_key_encryption_key` supports all masterkey formats.

**Review by kevinAlbs — APPROVED** [review]
LGTM

File: bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/CAPI.java [hunk 2]
```diff
@@ -829,6 +829,19 @@ public interface mongocrypt_random_fn extends Callback {
...
+     * Initialize a context to rewrap datakeys.
+     *
+     * Associated options {@link #mongocrypt_ctx_setopt_key_encryption_key(mongocrypt_ctx_t, mongocrypt_binary_t)}
+     *
+     * @param ctx The @ref mongocrypt_ctx_t object.
+     * @param filter The filter to use for the find command on the key vault collection to retrieve datakeys to rewrap.
+     * @return A boolean indicating success. If false, and error status is set.
+     * @since 1.5
+     */
+    public static native boolean
```
Comment (DmitryLukyanov) [inline bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/CAPI.java:842]: mongocrypt_ctx_setopt_key_material this will be a separate change?

**Review by DmitryLukyanov — APPROVED** [review]
Question and then LGTM

**Pull request merged**